src/filesys.cpp

```diff
@@ -758,14 +758,32 @@ bool safeWriteToFile(const std::string &path, const std::string &content)
 	std::string tmp_file = path + ".~mt";
 
 	// Write to a tmp file
+	bool tmp_success = false;
+
+#ifdef _WIN32
+	// We've observed behavior suggesting that the MSVC implementation of std::ofstream::flush doesn't
+	// actually flush, so we use win32 APIs.
+	HANDLE tmp_handle = CreateFile(
+		tmp_file.c_str(), GENERIC_WRITE, 0, nullptr, CREATE_ALWAYS, FILE_ATTRIBUTE_NORMAL, nullptr);
```

Comment:
could maybe use `FILE_ATTRIBUTE_HIDDEN` here too, but dunno if there are wider implications

Reply:
Seems totally orthogonal to what I'm trying to accomplish here, so I'd rather not in this PR.